[EXPO SDK53] extraMavenRepos not applied on gradle subprojects

### Minimal reproducible example

https://github.com/janwiebe-jump/sdk-53-extramavenrepos

### What platform(s) does this occur on?

Android

### Where did you reproduce the issue?

in a development build

### Summary

Since SDK 53 the extraMavenRepos property of expo-build-properties is not added to the non-root gradle project.
In our case, react-native-reanimated is installed, and the notifee maven repo is not added to this gradle project

Checkout the repo, install dependencies and try run `npx expo run:android`


```
(base) janwiebe@MB-JW-2 test-53 % npx expo run:android  
✔ Created native directory
✔ Updated package.json | no changes
» android: EDGE_TO_EDGE_PLUGIN: No configuration found for `edgeToEdgeEnabled` field in the project app config, falling back to false. In Android 16+ (targetSdkVersion 36) it will no longer be possible to disable edge-to-edge. Learn more: https://expo.fyi/edge-to-edge-rollout
✔ Finished prebuild
› Building app...
To honour the JVM settings for this build a single-use Daemon process will be forked. For more on this, please refer to https://docs.gradle.org/8.13/userguide/gradle_daemon.html#sec:disabling_the_daemon in the Gradle documentation.
Daemon will be stopped at the end of the build 
Configuration on demand is an incubating feature.

> Configure project :
Adding extra maven repository: ../../node_modules/@notifee/react-native/android/libs
[ExpoRootProject] Using the following versions:
  - buildTools:  35.0.0
  - minSdk:      24
  - compileSdk:  35
  - targetSdk:   35
  - ndk:         27.1.12297006
  - kotlin:      2.0.21
  - ksp:         2.0.21-1.0.28

> Configure project :notifee_react-native
:notifee_react-native @notifee/react-native found at /Users/janwiebe/projects/Temp/test-53/node_modules/@notifee/react-native
:notifee_react-native package.json found at /Users/janwiebe/projects/Temp/test-53/node_modules/@notifee/react-native/package.json
:notifee_react-native:version set from package.json: 9.1.8 (9,1,8 - 9001008)
:notifee_react-native:android.compileSdk using custom value: 35
:notifee_react-native:android.targetSdk using custom value: 35
:notifee_react-native:android.minSdk using custom value: 24
:notifee_react-native:reactNativeAndroidDir /Users/janwiebe/projects/Temp/test-53/node_modules/react-native/android

> Configure project :expo

Using expo modules
  - expo-constants (17.1.2)
  - expo-modules-core (2.3.5)
  - [📦] expo-asset (11.1.2)
  - [📦] expo-file-system (18.1.4)
  - [📦] expo-font (13.1.2)
  - [📦] expo-keep-awake (14.1.2)


> Configure project :react-native-reanimated
Android gradle plugin: 8.8.2
Gradle: 8.13

[Incubating] Problems report is available at: file:///Users/janwiebe/projects/Temp/test-53/android/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':app:processDebugResources'.
> Could not resolve all dependencies for configuration ':app:debugRuntimeClasspath'.
   > Could not find any matches for app.notifee:core:+ as no versions of app.notifee:core are available.
     Searched in the following locations:
       - file:/Users/janwiebe/projects/Temp/test-53/node_modules/react-native/android/app/notifee/core/maven-metadata.xml
       - https://dl.google.com/dl/android/maven2/app/notifee/core/maven-metadata.xml
       - https://repo.maven.apache.org/maven2/app/notifee/core/maven-metadata.xml
       - https://www.jitpack.io/app/notifee/core/maven-metadata.xml
       - https://oss.sonatype.org/content/repositories/snapshots/app/notifee/core/maven-metadata.xml
     Required by:
         project :app > project :notifee_react-native

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.13/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 29s
28 actionable tasks: 4 executed, 24 up-to-date
Error: /Users/janwiebe/projects/Temp/test-53/android/gradlew app:assembleDebug -x lint -x test --configure-on-demand --build-cache -PreactNativeDevServerPort=8081 -PreactNativeArchitectures=arm64-v8a exited with non-zero code: 1
Error: /Users/janwiebe/projects/Temp/test-53/android/gradlew app:assembleDebug -x lint -x test --configure-on-demand --build-cache -PreactNativeDevServerPort=8081 -PreactNativeArchitectures=arm64-v8a exited with non-zero code: 1
    at ChildProcess.completionListener (/Users/janwiebe/projects/Temp/test-53/node_modules/@expo/spawn-async/src/spawnAsync.ts:67:13)
    at Object.onceWrapper (node:events:633:26)
    at ChildProcess.emit (node:events:518:28)
    at maybeClose (node:internal/child_process:1101:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:304:5)
    ...
    at spawnAsync (/Users/janwiebe/projects/Temp/test-53/node_modules/@expo/spawn-async/src/spawnAsync.ts:28:21)
    at spawnGradleAsync (/Users/janwiebe/projects/Temp/test-53/node_modules/@expo/cli/src/start/platforms/android/gradle.ts:134:28)
    at assembleAsync (/Users/janwiebe/projects/Temp/test-53/node_modules/@expo/cli/src/start/platforms/android/gradle.ts:83:16)
    at runAndroidAsync (/Users/janwiebe/projects/Temp/test-53/node_modules/@expo/cli/src/run/android/runAndroidAsync.ts:62:24)
```

### Environment

```text
expo-env-info 1.2.2 environment info:
    System:
      OS: macOS 15.4
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 22.14.0 - ~/.nvm/versions/node/v22.14.0/bin/node
      Yarn: 1.22.22 - ~/.yarn/bin/yarn
      npm: 10.9.2 - ~/.nvm/versions/node/v22.14.0/bin/npm
      Watchman: 2025.03.10.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.15.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 24.4, iOS 18.4, macOS 15.4, tvOS 18.4, visionOS 2.4, watchOS 11.4
      Android SDK:
        API Levels: 33, 34, 35
        Build Tools: 30.0.3, 31.0.0, 32.0.0, 33.0.3, 34.0.0, 35.0.0
        System Images: android-26 | Google APIs ARM 64 v8a, android-28 | Google APIs ARM 64 v8a, android-30 | Google Play ARM 64 v8a, android-31 | Google Play ARM 64 v8a, android-32 | Google Play ARM 64 v8a, android-33 | Google APIs ARM 64 v8a, android-33 | Google Play ARM 64 v8a, android-34 | Google Play ARM 64 v8a, android-35 | Google Play ARM 64 v8a
    IDEs:
      Android Studio: 2024.2 AI-242.21829.142.2421.12409432
      Xcode: 16.3/16E140 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^53.0.0-preview.7 => 53.0.0-preview.7 
      react: 19.0.0 => 19.0.0 
      react-dom: 19.0.0 => 19.0.0 
      react-native: 0.79.0 => 0.79.0 
      react-native-web: ^0.20.0 => 0.20.0 
    npmGlobalPackages:
      eas-cli: 16.3.0
```

### Expo Doctor Diagnostics

```text
15/15 checks passed. No issues detected!
```

--- a/packages/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/SettingsManager.kt
+++ b/packages/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/SettingsManager.kt
@@ -112,11 +112,18 @@ class SettingsManager(
     }
 
     settings.gradle.beforeRootProject { rootProject: Project ->
-      config.allPlugins.forEach(rootProject::linkBuildDependence)
-      config.extraDependencies.forEach { mavenConfig ->
+      val extraDependency = config.extraDependencies
+      extraDependency.forEach { mavenConfig ->
         rootProject.logger.quiet("Adding extra maven repository: ${mavenConfig.url}")
-        rootProject.linkMavenRepository(mavenConfig)
+
       }
+      rootProject.allprojects { project ->
+        extraDependency.forEach { mavenConfig ->
+          project.linkMavenRepository(mavenConfig)
+        }
+      }
+
+      config.allPlugins.forEach(rootProject::linkBuildDependence)
 
       // Adds maven repositories for all projects that are using the publication.
       // It most likely means that we will add "https://maven.pkg.github.com/expo/expo" to the repositories.

--- a/packages/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/gradle/ProjectExtension.kt
+++ b/packages/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/gradle/ProjectExtension.kt
@@ -37,7 +37,7 @@ internal fun Project.linkMavenRepository(mavenRepo: MavenRepo) {
   val (url, credentials, authentication) = mavenRepo
 
   repositories.maven { maven ->
-    maven.url = URI.create(url)
+    maven.setUrl(url)
     maven.applyCredentials(credentials)
     maven.applyAuthentication(authentication)
   }

--- a/packages/expo-modules-core/CHANGELOG.md
+++ b/packages/expo-modules-core/CHANGELOG.md
@@ -8,7 +8,7 @@
 
 ### 🐛 Bug fixes
 
-- Fixed `extraMavenRepos` not applied on Gradle subprojects.
+- Fixed `extraMavenRepos` not applied on Gradle subprojects. ([#36500](https://github.com/expo/expo/pull/36500) by [@lukmccall](https://github.com/lukmccall))
 
 ### 💡 Others
 

--- a/packages/expo-modules-core/CHANGELOG.md
+++ b/packages/expo-modules-core/CHANGELOG.md
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- Fixed `extraMavenRepos` not applied on Gradle subprojects.
+
 ### 💡 Others
 
 ## 2.3.11 — 2025-04-28

